diff --git a/src/virtualkeyboard/virtualkeyboardview.cpp b/src/virtualkeyboard/virtualkeyboardview.cpp
--- a/src/virtualkeyboard/virtualkeyboardview.cpp
+++ b/src/virtualkeyboard/virtualkeyboardview.cpp
@@ -128,3 +128,7 @@ void VirtualKeyboardView::move(int x, int y) {
     view_->setX(x);
     view_->setY(y);
+    if (getDesktopType() == DesktopType::WAYLAND) {
+        LayerShellQt::Window::get(view_.get())
+            ->setMargins(QMargins(x, y, 0, 0));
+    }
     emit positionChanged(QPoint(x, y));
